### DISCUSSION TITLE
How can I autofill a Text Field conditionally based on an Option Picker field?

### DISCUSSION THREAD

**Discussion by EmonSM** [Budibase/budibase#8171] [Help] [answered]
I have a Budibase DB table with 2 columns: "Name" and "Address".

On my form, I have a name field that's an OPTION PICKER, which populates the options using a data provider that uses the "Name" column from the DB table.

My form also has an address field that's a TEXT FIELD. When a name is selected within the name option picker, I want the corresponding "Address" from that specific "Name"'s row in the table to be automatically filled within the address text field. 

How would I achieve this?

**Comment by andz-bb** [comment, marked as answer]
hey @EmonSM 

if the form is for a separate table to the one with the name and address you can add a Data Provider for the Name/Address table. Then you can add an Options Picker with the address field from the form table. In the Options Picker settings choose the Data Provider as the Options Source. Use Name for the Label Column and Address for the Value Column. If you have a separate address field in the form that's linked to the same value in the table it should auto update with the relevant address when you select a name in the picker.

![Screen Recording 2022-10-07 at 18 41 21](https://user-images.githubusercontent.com/110921612/194618003-120efb7b-0405-48e9-8c86-3bf9376adf54.gif)

Hope this make sense to you, if you have any questions let me know and I'll be happy to explain in more detail 👍

**Reply by EmonSM** [reply]
Hey @andz-bb

Thank you so much for your answer, that behaviour is exactly what I'm looking for.  I had a question though. You mentioned "If you have a separate address field in the form that's linked to the same value in the table, it should auto update with the relevant address when you select a name in the picker". How would I set up the address field so that it's "linked to the same value in the table"? Can you post a screenshot of your address field settings?

**Reply by andz-bb** [reply]
hi @EmonSM 

just to clarify, I meant that for both the name dropdown and the address field they should both use the same setting for Field. In my case I'm using the value `Delivery Address`. When you make a selection on the dropdown it will update the value on both components.

<img width="849" alt="image" src="https://user-images.githubusercontent.com/110921612/195072370-9e49285e-b49d-4fd4-b11b-712f1d5b1542.png">
<img width="849" alt="image" src="https://user-images.githubusercontent.com/110921612/195072523-3feffc66-ccca-41ee-a9d2-b808e2d47ed8.png">

**Reply by EmonSM** [reply]
Hi @andz-bb 

Ah, okay, I understand. I was able to get this scenario working, thank you! However, I actually need multiple fields to autofill based on name, not such one. I.e. My real data provider table has Name, Address, City, Zip Code, etc. columns. When a Name is selected in the option picker, I need all of the corresponding Address, City, Zip Code, etc. text fields to autofill. If I'm not mistaken, this current method of autofilling does not work with more than 1 field. Do you have any suggestions on how to autofill multiple different fields from one option being selected?
